libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSource.kt

```diff
@@ -56,21 +60,67 @@ private const val MINIMUM_ZOOM_LEVEL_FOR_GEO = 2.0
+ * Make sure to get familiar with all of the public nested fields and their documentation in
+ * [MapboxNavigationViewportDataSourceOptions]. The options can be mutated by accessing
+ * [options] field.
+ *
+ * Whenever any changes are made to the data source (new values provided, overrides added/removed,
+ * or options changed), remember to call [evaluate] recompute frames and notify observers.
+ *
+ * ## Padding and framing behavior
+ * This data source initializes at the `null island` (0.0, 0.0). Make sure to first provide at least
+ * [onLocationChanged] for following mode framing and [onRouteChanged] for overview mode framing
+ * (or the [additionalPointsToFrameForOverview] and [additionalPointsToFrameForFollowing]).
+ *
+ * ### Overview
+ * [overviewPadding] is used to generate the correct zoom level while positioning the contents on screen
+ * and is also applied to the resulting default [ViewportData.cameraForOverview].
+ * The default bearing for overview framing is north (`0.0`).
+ *
+ * ### Following
+ * [followingPadding] is used to generate the correct zoom level while positioning the contents on screen
+ * but the padding value is not applied the [ViewportData.cameraForFollowing].
+ * Instead, the frame contains a specific [CameraOptions.padding] value that manipulates the vanishing point of the camera
+ * to provide a better experience for end users when the camera is pitched.
+ *
+ * **This vanishing point change cannot be recovered from automatically without impacting the camera position.
+ * That's why, if you use the [MapboxNavigationViewportDataSource],
+ * you should explicitly define [CameraOptions.padding] in all other transition that your app is running.**
+ * The side-effect of not recovering from the vanishing point change can be the center of the camera that's offset from the center of the [MapView].
+ *
+ * **When following frame is used, the first point of the framed geometry list will be placed at the bottom edge of this padding, centered horizontally.**
+ * This typically refers to the user's location provided via [onLocationChanged], if available.
+ * This can be influenced with [FollowingFrameOptions.maximizeViewableRouteGeometryWhenPitchZero].
  *
- * ### Examples
+ * **The geometries that are below the bottom edge of the following padding on screen (based on camera's bearing) are ignored and not being framed.**
+ * It's impossible to find a zoom level that would fit geometries that are below the vanishing point of the camera,
+ * since the vanishing point is placed at the bottom edge of the provided padding.
+ *
+ * The default pitch for following frames is [FollowingFrameOptions.defaultPitch] and zoom is determined based on upcoming geometries or [FollowingFrameOptions.maxZoom].
+ *
+ * ## Debugging
+ * **This feature is currently experimental an subject to change.**
+ *
+ * You can use [debugger] to provide a [MapboxNavigationViewportDataSourceDebugger] instance
```

Comment:
Should `MapboxNavigationViewportDataSourceDebugger` be moved out the `internal` package?